app/View/Components/GraphRow.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace App\View\Components;
+
+use Illuminate\View\Component;
+
+class GraphRow extends Component
+{
+    /**
+     * @var string
+     */
+    public $type;
+    /**
+     * @var string
+     */
+    public $loading;
+    /**
+     * @var null
+     */
+    public $device;
+    /**
+     * @var null
+     */
+    public $port;
+    /**
+     * @var array|string[][]
+     */
+    public $graphs;
+    /**
+     * @var string|null
+     */
+    public $title;
+    /**
+     * @var float|int
+     */
+    public $rowWidth;
+    /**
+     * @var bool
+     */
+    public $responsive;
+    /**
+     * @var string
+     */
+    public $aspect;
+
+    /**
+     * Create a new component instance.
+     *
+     * @param  string  $type
+     * @param  string|null  $title
+     * @param  string  $loading
+     * @param  string  $aspect
+     * @param  int  $columns
+     * @param  array  $graphs
+     * @param  \App\Models\Device|int|null  $device
+     * @param  \App\Models\Port|int|null  $port
+     */
+    public function __construct(string $type = '', string $title = null, string $loading = 'eager', string $aspect = 'normal', int $columns = 2, array $graphs = [['from' => '-1d'], ['from' => '-7d'], ['from' => '-30d'], ['from' => '-1y']], $device = null, $port = null)
```

Comment:
@murrant 
int $columns = 2
Type hint doesn't make sense here, maybe string|int
But calculate Row Width except int also
Feels like it should be 2 different variables

Reply:
The reason I put them together is because you cannot set a column count in responsive mode.  Definitely not convinced that is correct, but it is where I settled on.